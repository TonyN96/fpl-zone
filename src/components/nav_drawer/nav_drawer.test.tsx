import React from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter as Router } from "react-router-dom";
import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";

import NavDrawer from "./nav_drawer";

describe("Nav Drawer Tests", () => {
  const mockQueryClient = new QueryClient();

  const templateRenderFunction = (label: string): JSX.Element => {
    return (
      <QueryClientProvider client={mockQueryClient}>
        <Router>
          <NavDrawer activeLabel={label} />
        </Router>
      </QueryClientProvider>
    );
  };

  it("Renders as expected", () => {
    const { asFragment } = render(templateRenderFunction("gameweek-live"));
    expect(asFragment()).toMatchSnapshot();
  });

  describe("Active label has expected styling", () => {
    it("When 'gameweek live' is active", () => {
      render(templateRenderFunction("gameweek live"));
      expect(screen.getByText("GAMEWEEK LIVE")).toHaveStyle("color: black");
      expect(screen.getByText("MY TEAM")).toHaveStyle("color: white");
    });

    it("When 'my team' is active", () => {
      render(templateRenderFunction("my team"));
      expect(screen.getByText("GAMEWEEK LIVE")).toHaveStyle("color: white");
      expect(screen.getByText("MY TEAM")).toHaveStyle("color: black");
    });
  });
});
