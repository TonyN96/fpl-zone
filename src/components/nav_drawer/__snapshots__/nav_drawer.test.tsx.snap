// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Drawer Tests Renders as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiDrawer-root MuiDrawer-docked css-750k1e-MuiDrawer-docked"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        class="logo-container MuiBox-root css-0"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-1ebt5mt-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <img
              alt="logo"
              class="football-icon"
              height="50px"
              src="/assets/images/football.png"
              width="50px"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-yu75kd-MuiGrid-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-body1 css-1ncgoc1-MuiTypography-root"
            >
              FPL Zone
            </h1>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-zf9gw9"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/gameweek-live"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                GAMEWEEK LIVE
              </span>
            </div>
          </a>
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/gameweek-live"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                MY TEAM
              </span>
            </div>
          </a>
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/fixtures-and-results"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                FIXTURES & RESULTS
              </span>
            </div>
          </a>
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/analysis"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                ANALYSIS
              </span>
            </div>
          </a>
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/social"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                SOCIAL
              </span>
            </div>
          </a>
        </ul>
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/logout"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                LOGOUT
              </span>
            </div>
          </a>
          <a
            class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-om9qhu-MuiButtonBase-root-MuiListItemButton-root"
            href="/account"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-17gt2ew-MuiTypography-root"
              >
                ACCOUNT
              </span>
            </div>
          </a>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
